### ISSUE TITLE
DJ mode

### ISSUE THREAD

**Issue by stoneubi** [fast4x/RiMusic#1778]
### Description

Thank you for the opportunity to create a feature request for https://github.com/fast4x/RiMusic/discussions/1748.

This feature request asks for a DJ mode in RiMusic. 

### Suggested Solution

Actually, if you have a playlist played and you search for a new song and you select this song, the song will start to play immediately and the playlist will be changed. 

Having DJ mode enabled RiMusic should work as follows:
* at the settings menu there should be an option called DJ mode, which can be activated to de-activated 
* plus there should be two options
* one option for fade in time and 
* one for when the fade in should start

For example, fade in time is 5 seconds and start time is 10 second would mean that 10 seconds before the actual played song is finished the next song will start to fade in and this fade in will take 5 seconds. 

Also with DJ mode enabled the behavior of RiMusic will be that if you have already a playlist playing and songs are in the playlist and you are e.g. at a party or event and a guest will come over and ask for a favorite song, you search for that song e.g. online and once you select this song, the song will not start playing right away, the song will be added to the existing playlist as the next song. This is like the actual feature called "Play next" but you don't have to always select "Play next". This will be done automatically if DJ mode is enabled and you don't have the risk to search for a song and to forget to manually select this option. 

Wow, this would be so great. 

Thank you very much for making this possible,
Stoneubi

**Comment by JeePopuwu** [comment]
It sounds very engaging and interesting.  I would like to see that

**Comment by AquamarineDreams** [comment]
That seems really interesting, it would be really useful. And, if you make the "auto play next" as a toggle, you would have a very similar feature to Spotify's gradual song transition.

**Comment by aneesh1122** [comment]
I like the implementation of Spotify. Just right swipe the song and it will be added to 'play next'

**Comment by cMattw** [comment]
> I like the implementation of Spotify. Just right swipe the song and it will be added to 'play next'

This is already in the app, it's just different. Instead of play next, it's add to queue. Which adds it to the bottom of the queue. If the dev could replace that button it would be great

**Comment by AquamarineDreams** [comment]
Yes, but it only works on a user-created playlist. It would be nice to have this feature in albums or searches.

**Comment by AquamarineDreams** [comment]
> > I like the implementation of Spotify. Just right swipe the song and it will be added to 'play next'
> 
> This is already in the app, it's just different. Instead of play next, it's add to queue. Which adds it to the bottom of the queue. If the dev could replace that button it would be great

Maybe this could be changed in the settings. Play next as default, and add to queue as an option.

**Comment by aneesh1122** [comment]
> > I like the implementation of Spotify. Just right swipe the song and it will be added to 'play next'
> 
> This is already in the app, it's just different. Instead of play next, it's add to queue. Which adds it to the bottom of the queue. If the dev could replace that button it would be great

Spotify adds the song to 'play next' just after the right swipe.

Here you have to swipe and then select 'add to queue'

Spotify has only one step whereas RiMusic requires 2 steps.

**Comment by stoneubi** [comment]
I don't use spotify at all and therefore I event didn't know about the existing feature. I'm using newpipe and rimusic. My intention was, that using rimusic during my parties I prepare some playlists. During this good time it happened that I search for a song and instantly selected the song and bumm, I interrupted the playing song and changed my playlist. 

Therefore I started this feature request because I thought I don't might be the only user who acted like this and therefore the DJ mode might be a very good solution. 

I'm looking forward having this feature in RiMusic :-)

**Comment by aneesh1122** [comment]
> I don't use spotify at all and therefore I event didn't know about the existing feature. I'm using newpipe and rimusic. My intention was, that using rimusic during my parties I prepare some playlists. During this good time it happened that I search for a song and instantly selected the song and bumm, I interrupted the playing song and changed my playlist.
> 
> Therefore I started this feature request because I thought I don't might be the only user who acted like this and therefore the DJ mode might be a very good solution.
> 
> I'm looking forward having this feature in RiMusic :-)

@fast4x you can implement it this way

1) after selecting the song, add to the queue and start playing
2) right swipe to add it to 'play next' in the queue
This is how it's done in spotify

I don't know who would want the queue to get cleared after selecting a song so just remove it.
